### PR TITLE
oci: blobs storage is blobs/{algo}/{hash}

### PR DESCRIPTION
**Pull request by runcom**
Fix https://github.com/containers/image/issues/66

@mtrmac PTAL

Signed-off-by: Antonio Murdaca <runcom@redhat.com>

### CODE REVIEW

**Comment by runcom** [comment]
I'll fix skopeo asap

File: oci/oci_transport.go [hunk 1]
```diff
@@ -193,7 +193,11 @@ func (ref ociReference) ociLayoutPath() string {
 
 // blobPath returns a path for a blob within a directory using OCI image-layout conventions.
 func (ref ociReference) blobPath(digest string) string {
-	return filepath.Join(ref.dir, "blobs", strings.Replace(digest, ":", "-", -1))
+	pts := strings.SplitN(digest, ":", 2)
+	if len(pts) != 2 {
+		return ""
+	}
```
Comment (mtrmac) [inline oci/oci_transport.go]: Doesn’t this need error handling in the callers?

Comment (mtrmac) [inline oci/oci_transport.go]: … and a verbose error message (say "Internal error: unexpected digest reference”) would be nice.

Comment (runcom) [inline oci/oci_transport.go]: I didn't want to change `blobPath` to return (error, string) but yeah it makes sense.. I'll take another pass on this later today

**Comment by runcom** [comment]
@mtrmac PTAL

**Comment by mtrmac** [comment]
LGTM, thanks (especially for the test keeping `oci_transport.go` coverage high.)

**Comment by runcom** [comment]
lgtm

**Pull request merged**